src/pybind/cephfs/setup.py

```diff
@@ -72,7 +72,7 @@ def get_python_flags(libs):
         libraries=libs + py_libs,
         extra_compile_args=filter_unsupported_flags(
```

Comment:
ACK: `-fstack-clash-protection` came from the [`new_compiler()`](https://docs.python.org/2/distutils/apiref.html#distutils.ccompiler.new_compiler) call.
`filter_unsupported_flags()` just squeezes some compiler options if it detects `clang`.

```python
def filter_unsupported_flags(compiler, flags):
    args = takewhile(lambda argv: not argv.startswith('-'), [compiler] + flags)
    if any('clang' in arg for arg in args):
        return list(filterfalse(lambda f:
                                f in ('-mcet',
                                      '-fstack-clash-protection',
                                      '-fno-var-tracking-assignments',
                                      '-Wno-deprecated-register',
                                      '-Wno-gnu-designator') or
                                f.startswith('-fcf-protection'),
                                flags))
    else:
        return flags
```